core/shared/src/main/scala/zio/internal/FiberContext.scala

```diff
@@ -252,10 +253,9 @@ private[zio] final class FiberContext[E, A](
               // Cannot capture `curZio` since it will be boxed into `ObjectRef`,
               // which destroys performance. So put `curZio` into a temp val:
               val tmpIo = curZio
-
-              curZio = ZIO.yieldNow *> tmpIo
-
-              opcount = 0
+              pushContinuation(_ => tmpIo)
+              evaluateLater(ZIO.Resume)
```

Comment:
Could `tmpIo` be evaluated later instead of introducing a new tag?

Reply:
The point is that we need to handle the parameter to `evaluateLater` before the while loop. I'd be happy about a solution without a special tag, but I couldn't think of one.

Reply:
why does the parameter to `evaluateLater` need to be handled before the while loop?

Reply:
Because we must not increase `maxopcount`. Otherwise we need to immediately yield again.

Reply:
I was thinking of something like:

```
if (opcount == maxopcount) {
    // Cannot capture `curZio` since it will be boxed into `ObjectRef`,
    // which destroys performance. So put `curZio` into a temp val:
    val tmpIo = curZio
    evaluateLater(tmpIo)
    curZio = null
} else {
```

With `curZio` set to `null` the while loop stops so we don't need to worry about opcount.

Reply:
Another idea is to add a `resume` to `Yield` and create a new `Yield` if `opcount == maxopcount` before matching on the tag. This also allows the addition of `yieldNow` to the trait so users can opt-in to yielding without creating an intermediate FlatMap.

https://github.com/zio/zio/compare/master...dreverri:djr/yield-resume?w=1

Reply:
I see. I like both ideas. I tend to prefer the first, because it is simpler. What do you think?

Reply:
Incorporated the first one. Thanks for the good suggestion. 🙏